diff --git a/pal/src/pal_main.c b/pal/src/pal_main.c
index 4586b39f..b78ee1b2 100644
--- a/pal/src/pal_main.c
+++ b/pal/src/pal_main.c
@@ -602,6 +602,9 @@ noreturn void pal_main(uint64_t instance_id,       /* current instance id */
 
     pal_disable_early_memory_bookkeeping();
 
+    uint64_t exec_time;
+    _PalSystemTimeQuery(&exec_time);
+    log_error("Time: %ld", exec_time);
     /* Now we will start the execution */
     start_execution(arguments, final_environments);
 
