diff --git a/tasks/qemu.py b/tasks/qemu.py
index b7195cf..9c822db 100644
--- a/tasks/qemu.py
+++ b/tasks/qemu.py
@@ -295,12 +295,12 @@ def spawn_qemu(
         qmp_socket = Path(tempdir).joinpath("qmp.sock")
         cmd = extra_args_pre.copy()
 
-        if numa_node is not None:
-            cmd += [
-                "numactl",
-                f"--cpunodebind={','.join(map(str, numa_node))}",
-                f"--membind={','.join(map(str, numa_node))}",
-            ]
+        #if numa_node is not None:
+        #    cmd += [
+        #        "numactl",
+        #        f"--cpunodebind={','.join(map(str, numa_node))}",
+        #        f"--membind={','.join(map(str, numa_node))}",
+        #    ]
 
         qmp_command = [
             "-qmp",
@@ -346,6 +346,7 @@ def spawn_qemu(
                     time.sleep(0.1)
                 except ProcessLookupError:
                     raise Exception("qemu vm was terminated")
+            subprocess.run(["sudo", "chmod", "0777", qmp_socket])
             with connect_qmp(qmp_socket) as session:
                 yield QemuVm(session, tmux_session, qemu_pid, config)
         finally:
